make/git.mk

```diff
@@ -1,4 +1,4 @@
-RELEASE_VERSION := $(shell git describe --tags)
+RELEASE_VERSION := $(shell git describe --tags --match='v*' --abbrev=14)
```

Comment:
Oh my god this sent me down a rabbit hole :joy: 

The issue is that the `v1.7.0` and `v1.7.1` tags aren't on master, they're on the `release-1.7` branch.

`git tag --list` shows _all tags on all branches_

`git describe --tags` shows _all tags on the current branch_. So if you're on master (or a feature branch based on master), it'll only include tags on master... in this case, the latest tag on master is v1.7.0-beta.0. 

```console
$ git tag --list | sort -r | head -n5
v1.7.1
v1.7.0-beta.0
v1.7.0-alpha.1
v1.7.0-alpha.0
v1.7.0

$ git checkout master
$ git describe --tags
v1.7.0-beta.0-141-geccd68253

$ git checkout release-1.7
$ git describe --tags
v1.7.1
```

You can see the tags which are reachable on a given branch using `git log --oneline --decorate=short | grep 'tag\:'`.

I agree that v1.7.1-something would be less confusing, but I don't really know how it's possible to do because v1.7.1 isn't a tag on master; I think except for the tag name being weird `git describe --tags` is a pretty good thing to use here.

Reply:
> I'm also confused as to why v1.7.1 is not on master whilst v1.7.0-beta.0 is.

```console
$ git merge-base master release-1.7
39e388eaa5b19b34d5a23e4a87b3e57812e111d1
```

That hash is the commit at which `master` and `release-1.7` diverged and split into two different histories. We continued committing to master after that hash, and we cherry-picked into `release-1.7` after that too. 

`git log --graph --oneline` also kinda helps for visualising it but it's a bit messy!